EIPS/eip-2027.md

```diff
@@ -0,0 +1,74 @@
+- If previous value of the [`location`] is 0, and value is not 0, *increment* `storagesize` (semantics of *increment* described below)
+- If previous value of the [`location`] is not 0, and value is 0, *decrement* `storagesize` (semantics of *decrement* described below)
+- As with other state changes, changes of `storagesize` get reverted when the execution frame reverts, i.e. it needs to use the same techniques as storage values, like journalling (in Geth), and substates (in Parity).
+Value of `storagesize` is not observable from contracts at this point.
+
+### Semantics of *increment* `storagesize`
+If `storagesize` is not present, `storagesize` = `HUGE_NUMBER` + 1.
+If `storagesize` is present, `storagesize` = `storagesize` + 1.
+
+### Semantics of *decrement* `storagesize`
+If `storagesize` is not present, `storagesize` = `HUGE_NUMBER` - 1.
+If `storagesize` is present, `storagesize` = `storagesize` - 1.
+
+### Note of `HUGE_NUMBER`
+There is a constant `HUGE_NUMBER`. It needs to be large enough so that no real metrics (contract storage size, number of accounts, number of contracts, total size of code, total size of storage) will never reach that number, and small enough that it fits in an unsigned 64-bit integer.
+Current suggestion is to have `HUGE_NUMBER` = 2^63, which is binary representation is the a single bit in a 64-bit number.
+
+The idea is to make it decidable later whether the storagesize was ever incremented/decremented (presence of the field), and whether it has been converted from net to gross (by value being smaller than `HUGE_NUMBER/2` - because it will not be possible for any contract be larger than 2^62 at the block C).
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+A mechanism for estimation of contract storage size has been proposed [here](https://medium.com/@akhounov/estimation-approximate-of-the-size-of-contracst-in-ethereum-4642fe92d6fe). But it does have a big drawback of introducing a lot of complexity into the consensus
+(in the form of estimation algorithm, which has quite a few edge cases to cater for different sizes of the storage).
+
+## Backwards Compatibility
+<!--All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+This change is not backwards compatible and requires hard fork to be activated.
+
```

Comment:
I would mention that since this new field is not observable and it doesn't change anything, it has no effect on contracts (on its own) -- but could with subsequent changes.

Reply:
Done, thank you